test/framework/src/main/java/org/elasticsearch/test/ESSingleNodeTestCase.java

```diff
@@ -367,4 +403,105 @@ protected boolean forbidPrivateIndexSettings() {
         return true;
     }
 
+    /**
```

Comment:
This doesn't feel like a great place for these test methods. We do have a `FieldMapperTestCase` but we don't use it consistently.